plots/laws/gravity/throwing_angle.py

```diff
@@ -0,0 +1,105 @@
+## (-g is the projection of free fall acceleration which is downwards to the Y axis which is upwards, so angle between acceleration and Axis is 180°)
+## The horizontal initial speed is projection of throwing_velocity to X, acceleration is 0.
+## Flight ends when y == 0 again.
+
+initial_horizontal_velocity = solve(projector.law, projector.projection, dict=True)[0][projector.projection].subs(
+    {
+        projector.vector_length: throwing_velocity, projector.vector_angle: throwing_angle
+    })
+print(f"Initial horizontal velocity: {initial_horizontal_velocity}")
+
+# the angle between initial velocity and Y axis is (pi/2 - throwing angle) (that's because of pi/2 angle between X and Y axis)
+initial_vertical_velocity = solve(projector.law, projector.projection, dict=True)[0][projector.projection].subs(
+    {
+        projector.vector_length: throwing_velocity, projector.vector_angle: pi / 2 - throwing_angle
+    })
+
+print(f"Initial vertical velocity: {initial_vertical_velocity}")
+
+x_function = movement_law.law.rhs.subs(
+    {        
+        movement_law.initial_velocity: initial_horizontal_velocity, 
+        movement_law.movement_time: time_argument, 
+        movement_law.constant_acceleration: 0
+    })
+print(f"x(t): {x_function}")
+
+y_function = movement_law.law.rhs.subs(
+    {         
+        movement_law.movement_time: time_argument, 
+        movement_law.constant_acceleration: -gravitational_acceleration,
+        movement_law.initial_velocity: initial_vertical_velocity        
+    })
+print(f"y(t): {y_function}")
+
+end_of_flight = Eq(0, y_function)
+# there are 2 points of the trajectory with h = 0, the first one is the start point, and the second one is at destination
+# We need the second one
+flight_time = solve(end_of_flight, time_argument, dict=True)[1][time_argument]
+print(f"Flight time: {flight_time}")
+
+flight_distance = x_function.subs({time_argument: flight_time})
+print(f"Flight distance: {flight_distance}")
+
+# for plotting purposes we don't care about scale and may substitute constants with 1
+flight_distance_plotted = flight_distance.subs(
+    {
+        throwing_velocity: 1,
+        gravitational_acceleration: 1
+    })
+flight_distance_plotted = simplify(flight_distance_plotted)
+print(f"Plotted flight distance: {flight_distance_plotted}")
+
+# Let's find needed angle analytically.
+# We are to find maximum of flight_distance(throwing_angle) function with angle in (0, pi/2) interval.
+# So the flight distance is the function of angle like k*cos(angle)*cos(pi/2 - angle).
+# cos(pi/2 - angle) is sin(angle), and sin(angle)*cos(angle) is 0.5*sin(2 * angle)
+flight_distance_manually_simplified = sin(2 * throwing_angle)
+distance_diff = Derivative(flight_distance_manually_simplified, throwing_angle).doit()
+max_law = Eq(0, distance_diff)
+max_angle = solve(max_law, throwing_angle, dict=True)[0][throwing_angle]
+print(f"Angle to achieve maximum distance: {max_angle}")
+
+Len = plot(
```

Comment:
Why Len? Not a big deal, but looks not better than our default `p0` name for plot.